compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/codegen/split/creators/JvmRecordCreatorGen.java

```diff
@@ -152,14 +158,29 @@ private void generateCreateRecordMethodSplits(ClassWriter cw, List<BIRTypeDefini
                         bTypesCount, remainingCases, labels, defaultCaseLabel);
                 i = 0;
             }
-            String fieldName = getTypeFieldName(optionalTypeDef.internalName.value);
             Label targetLabel = targetLabels.get(i);
             mv.visitLabel(targetLabel);
             mv.visitVarInsn(ALOAD, 0);
-            String className = getTypeValueClassName(moduleId, optionalTypeDef.internalName.value);
-            mv.visitTypeInsn(NEW, className);
-            mv.visitInsn(DUP);
-            mv.visitFieldInsn(GETSTATIC, moduleInitClass, fieldName, GET_TYPE);
+
+            BTypeSymbol referredTypeSymbol = JvmCodeGenUtil.getReferredType(optionalTypeDef.type).tsymbol;
+            String className;
+            if (referredTypeSymbol != null) {
+                className = getTypeValueClassName(referredTypeSymbol.pkgID, referredTypeSymbol.name.getValue());
+                mv.visitTypeInsn(NEW, className);
+                mv.visitInsn(DUP);
+                BType typeDefType = optionalTypeDef.type;
+                if (typeDefType.tag == TypeTags.TYPEREFDESC) {
+                    this.jvmTypeGen.loadType(mv, optionalTypeDef.referenceType);
+                } else {
+                    this.jvmTypeGen.loadType(mv, optionalTypeDef.type);
+                }
+            } else {
+                className = getTypeValueClassName(moduleId, optionalTypeDef.internalName.value);
+                mv.visitTypeInsn(NEW, className);
+                mv.visitInsn(DUP);
+                String fieldName = getTypeFieldName(optionalTypeDef.internalName.value);
+                mv.visitFieldInsn(GETSTATIC, moduleInitClass, fieldName, GET_TYPE);
```

Comment:
This is how this always was previously. Now it seems like, these are not getting covered. Will remove this if the full build pipeline passes without this.

Reply:
Removed the uncovered lines.